nfc/libs/NfcCoreLib/lib/Nci/phNciNfc.c

```diff
@@ -344,22 +344,28 @@ NFCSTATUS phNciNfc_StartDiscovery(void* pNciHandle,
             bNoofConfigs++;
         }
         /* Check whether Polling loop to be enabled for Listen NFC-F Technology */
-        if(pPollConfig->ListenNfcAActive)
+        if(pPollConfig->ListenNfcAActive ||
+            ((1 == pPollConfig->ListenNfcFActive && ((pNciContext->ResetInfo.NciVer & PH_NCINFC_VERSION_MAJOR_MASK) ==
+                                                     (PH_NCI2xNFC_VERSION & PH_NCINFC_VERSION_MAJOR_MASK)))))
```

Comment:
The major version is checked 4 times. It might be worth putting this into its own variable.